pkg/client/restconfig/restconfig.go

```diff
@@ -39,18 +39,24 @@ func NewRestConfig(timeout time.Duration) (*rest.Config, error) {
 	var cfg *rest.Config
 	// Detect kubectl config file
 	path, err := KubeConfigPath()
-	if err != nil {
-		// Build from k8s downward API
-		cfg, err = NewFromInClusterConfig()
-		if err != nil {
-			return nil, fmt.Errorf("failed to build rest config: kubeconfig not found: reading in-cluster config: %w", err)
+	kubeConfigExists := false
```

Comment:
Is there any way we can test this code? Looks right, but it'd be nice to have this little caveat tested to avoid regression.

Reply:
Added a unit test

Reply:
Sorry to give you a hard time about this, but I really don't love the pattern of using package private variables for test hooks unless there's no other way to do it. It feels invasive, is harder to read, and doesn't provide for extension.

Is there some way we could do this with dependency injection, and then just make a thin shim for` NewRestConfig ()`that calls like `NewRestConfigWithX(...)` that takes factory functions or a StatFunc or something?

Reply:
refactored

Reply:
Thanks!
I have some more nits, but I'll approve as-is for now.

- In the future maybe we could consider using a fake filesystem. There's a few available. Maybe k8s has one too. 
- I've also seen fake environments with methods for vars that make it easier to test things that use env vars. Might be useful if we want to test the full behavior and not just the delegation to upstream functions.
- Generally I think I prefer the convention of using -Fn for variables, -Func for types, and drop the suffix for struct fields, as if they were a normal method.
- Since there's no Build method, this isn't really a builder pattern. It's more like a factory pattern. But since we don't expose an interface, it doesn't really matter.